dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/CDIContainerImpl.java

```diff
@@ -172,20 +176,26 @@ public Void run() {
                 webSphereCDIDeployment.validateJEEComponentClasses();
                 weldBootstrap.deployBeans();
                 weldBootstrap.validateBeans();
-                return webSphereCDIDeployment;
             } else {
                 if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
                     Tr.debug(tc, "startInitialization", "CDI is not enabled, shutting down CDI");
                 }
                 webSphereCDIDeployment.shutdown();
                 unsetDeployment(application);
-                return null;
+                webSphereCDIDeployment = null;
             }
 
+        } catch (DeploymentException e) {
+            if (webSphereCDIDeployment != null) {
+                CDIContainerEventManager eventManager = this.cdiRuntime.getCDIContainerEventManager();
+                if (eventManager != null) {
+                    eventManager.processDeploymentException(webSphereCDIDeployment, e);
+                }
+            }
```

Comment:
Does this block need to re-throw the `DeploymentException` ?

Reply:
ah yeah, it was there before but i missed the else block when i moved the "process" part out to a different service